packages/nextjs/pages/blockexplorer.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { BlockWithTransactions, TransactionResponse } from "@ethersproject/abstract-provider";
+import { ethers } from "ethers";
+import type { NextPage } from "next";
+import { ArrowLeftIcon, ArrowRightIcon } from "@heroicons/react/24/outline";
+import { SearchBar } from "~~/components/blockexplorer/SearchBar";
+import { TransactionsTable } from "~~/components/blockexplorer/TransactionsTable";
+import deployedContracts from "~~/generated/deployedContracts";
+import { getTargetNetwork } from "~~/utils/scaffold-eth";
+
+// TODO: Add OPCODES and maybe STORAGE for contracts
+// TODO: Better pagination(some pages have less than BLOCKS_PER_PAGE blocks)
+// TODO: Refactor, seperate the table and the search bar to a component
+// TODO: Don't do everything in a loop... Please...
+// TODO: try useMemo to avoid unnecessary re-renders
+// TODO: Error handling for API calls
+// TODO: Add tests
+
+type TransactionWithFunction = TransactionResponse & { functionName?: string };
+
+const BLOCKS_PER_PAGE = 20;
+
+const Blockexplorer: NextPage = () => {
+  const [blocks, setBlocks] = useState<BlockWithTransactions[]>([]);
```

Comment:
We should probably export type from -> https://github.com/scaffold-eth/scaffold-eth-2/pull/351/files#r1199794750 to use it for state `Block[]`
trying to be more consistent